lib/cocoapods/installer/xcode/pods_project_generator.rb

```diff
@@ -198,10 +198,8 @@ def set_target_dependencies
               configure_app_extension_api_only_for_target(aggregate_target) if is_app_extension
 
               unless pod_target.should_build?
-                pod_target.resource_bundle_targets.each do |resource_bundle_target|
-                  aggregate_target.native_target.add_dependency(resource_bundle_target)
-                end
-
+                add_resource_bundles_to_native_target(pod_target, aggregate_target.native_target)
+                add_pod_target_test_dependencies(pod_target, frameworks_group)
                 next
```

Comment:
This `next` was causing to ignore integrating the test native targets for any pod target. We now ensure we do the right thing even in pods that they should not be built.